src/CUDA/CUDAallocator.hpp

```diff
@@ -154,6 +154,41 @@ bool operator!=(const CUDAHostAllocator<T1>&, const CUDAHostAllocator<T2>&)
 {
   return false;
 }
+
+/// allocator locks memory pages allocated by HostAllocator
+template<typename T, class HostAllocator = std::allocator<T>>
```

Comment:
Here is where I think the naming is an issue. Somehow I lost this comment so my final review didn't make much sense.
the Template parameter HostAllocator seems like it has a connection to the CudaHostAllocator  immediately above, but this is in fact an allocator that wouldn't be used with CUDALockedPageAllocator.  Changing the parameter to say `ULHA` (I like short all caps template param names) and defining that as an unlocked host allocator was what I was thinking.

Reply:
Renamed template argument and add more documentation.